### ISSUE TITLE
Fields to point to help and hint with `aria-describedby`

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#4832]
- The XForms should
    - [ ] Always produce an `id` on the help and hint elements
    - [ ] When we have a help or a hint, produce on the field an `aria-describedby` pointing to the help/hint ids.

[+1 from customer](https://3.basecamp.com/3600924/buckets/2016476/messages/3528991688)

**Comment by ebruchez** [comment]
The help element `id` needs to be there when we know that we are going to point to it.

Not only native form fields can use this attribute, per MND:

> The `aria-describedby` attribute is not limited to form controls. It can also be used to associate static text with widgets, groups of elements, regions that have a heading, definitions, and more. The aria-describedby attribute can be used with semantic HTML elements and with elements that have an ARIA [role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles).

**Comment by ebruchez** [comment]
Interestingly, we already have [code](https://github.com/orbeon/orbeon-forms/blob/4421f85a8e4f9961da12691b69f30ac9a1610244/xforms-runtime/shared/src/main/scala/org/orbeon/oxf/xforms/processor/handlers/xhtml/XFormsControlLifecyleHandler.scala#L217) to do this. We map:

- `aria-describedby` to hint
- `aria-details` to help

So why doesn't this appear to work?

**Comment by ebruchez** [comment]
It's not working because we only trigger that code within repeats for external labels. Also, we don't yet add ids on `hint` and `help` elements.

**Comment by ebruchez** [comment]
Adding the above helps with Voice Over. However `aria-details` is not read by Voice Over when on the field, although it is read when tabbing through the help (now that it is tabbable). Should we make both help and hint `aria-describedby`? The problem with that might be that `aria-describedby`/help can be very long.

**Comment by ebruchez** [comment]
Also for example the Image Attachment control doesn't link `aria-describedby`/`aria-details` AFAICT.

**Comment by ebruchez** [comment]
Per MDN:

> The `aria-details` attribute can be used to provide additional information or complex descriptions to an object. 

In theory at least, using that for help would make sense, unless help is short.

However support for `aria-details` appears flimsy in screen readers. This is unfortunate as that's the perfect solution for help, and if this was supported we could even skip tabbing through the help icon.

So we could still use `aria-describedby` for help (in addition to hint). But then the user will be read a potentially long message, which, granted, they should be able to stop.

Alternatively, we could have `aria-describedby` point to a message that says "Help is available, do this to access the help message".

**Comment by ebruchez** [comment]
Quick summary. To link a LHHA to a control, two cases:

1. `label` AND target is an HTML form element
  - in this case, use `for` ("forward" linking)
2. target is not an HTML form element OR HHA
  - in this case, use `aria-labelledby`/`aria-describedby` ("backward" linking)
  - for `help` too, use `aria-labelledby` (either generic message or help message) as `aria-details` is not supported

Currently, we only support "forward" linking.

For labels, in some cases we have used `aria-label` to work around the lack of "backward" linking. But this duplicates the label in the DOM and increases the amount of expressions/computations to make.

Steps:

- [ ] `label`: when target is NOT an HTML form element, output `aria-labelledby` instead (IFF no `aria-label` already)
- [ ] add support for `aria-describedby`
- [ ] TBD
- [ ] remove existing uses of `aria-label`

**Comment by ebruchez** [comment]
For example with `fr:open-select1`, we generate internally:

```xml
<xh:span class="fr-component-group" role="group" aria-label="{{xxf:label('fr-open-select1')}}">
```

Since the binding doesn't have `xxbl:label-for`, and right now the label and hint are orphan `<span>`s, the default behavior should be to add `aria-labelledy`/`aria-describedby` on the component's containing element.

**Comment by ebruchez** [comment]
I have a bug whereby updating the external LHHA for a control fails. This might be a regression or not, but it might be a case that is not used by Form Builder except for the case of external repeat controls, which is a little different. The repeat case works, but not the regular external case.

What's happening in both cases:

- external:
  - markup: has id `my-input≡≡t`
  - update response
    - has `xf-3`, which is the effective id of the `XFormsLHHAControl`
    - in addition, adds `<xxf:control id="my-input" label="My new label"/>`, which shouldn't be here
- repeat:
  - markup: has id `xf-5`
  - update response: has `xf-5`, which is the effective id of the `XFormsLHHAControl`, and is correct!

Looking at `XFormsLHHAHandler`, what are the differences between the two cases?

- external
  -  uses `handleLabelHintHelpAlert()`
  - supports constraint classes, `xforms-disabled`
- repeat
  - uses `withElement()` directly

Ajax response:

- in `ControlAjaxSupport`,
- see also  #3836
- filters with `staticLhhaSupport.hasLHHANotForRepeat(lhha)`

**Comment by ebruchez** [comment]
Difference in the markup produced so far:

- ids present on help/hint when needed
- `labelled-by`/`described-by` present on HTML form controls when needed
